src/solvers/refinement/string_refinement_util.cpp

```diff
@@ -322,18 +322,20 @@ std::vector<mp_integer> string_insertion_builtin_functiont::eval(
   const std::vector<mp_integer> &args_value) const
 {
   PRECONDITION(args_value.size() >= 1 || args_value.size() <= 3);
-  const std::size_t &offset = numeric_cast_v<std::size_t>(args_value[0]);
-  const std::size_t &start =
-    args_value.size() > 1 ? numeric_cast_v<std::size_t>(args_value[1]) : 0;
-  const std::size_t &end = args_value.size() > 2
-                             ? numeric_cast_v<std::size_t>(args_value[2])
-                             : input2_value.size();
+  const auto offset = max(args_value[0], 0);
+  const auto start =
+    max(args_value.size() > 1 ? args_value[1] : mp_integer(0), 0);
+
+  const auto end = max(
+    min(args_value.size() > 2 ? args_value[2] : mp_integer(input2_value.size()),
+        input2_value.size()),
+    0);
```

Comment:
Can't we just use `std::min` and `std::max`? Or really rewrite the entire code?
````c++
std::size_t offset = 0, start = 0, end = 0;
if(args_value[0] > 0)
  offset = integer2size_t(args_value[0]);

if(args_value.size() > 1 && args_value[1] > 0)
  start = integer2size_t(args_value[1]);

if(args_value.size() > 2 && args_value[2] > 0)
  end = std::min(integer2size_t(args_value[2]), input_value.size());
````